File: src/Baekjoon/p10845.java
package Baekjoon;

import java.io.*;

public class p10845 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder bw = new StringBuilder();

        int N = Integer.parseInt(br.readLine());
        Queue queue = new Queue();

        for (int i = 0; i < N; i++) {
            String[] command = br.readLine().split(" ");

            if (command[0].equals("push")) {
                int x = Integer.parseInt(command[1]);
                queue.push(x);
            }
            else if (command[0].equals("pop")) {
                bw.append(queue.pop()).append('\n');
            }
            else if (command[0].equals("size")) {
                bw.append(queue.size()).append('\n');
            }
            else if (command[0].equals("empty")) {
                bw.append(queue.empty()).append('\n');
            }
            else if (command[0].equals("front")) {
                bw.append(queue.front()).append('\n');
            }
            else if (command[0].equals("back")) {
                bw.append(queue.back()).append('\n');
            }
        }

        System.out.print(bw);
    }

    private static class Queue {
        private int[] queue;
        private int frontIndex;
        private int backIndex;

        public Queue() {
            queue = new int[10000];
            frontIndex = 0;
            backIndex = -1;
        }

        public void push(int x) {
            queue[++backIndex] = x;
        }

        public int pop() {
            if (empty() == 1) {
                return -1;
            }
            return queue[frontIndex++];
        }

        public int size() {
            return backIndex - frontIndex + 1;
        }

        public int empty() {
            if (backIndex < frontIndex) {
                return 1;
            }
            return 0;
        }

        public int front() {
            if (empty() == 1) {
                return -1;
            }
            return queue[frontIndex];
        }

        public int back() {
            if (empty() == 1) {
                return -1;
            }
            return queue[backIndex];
        }
    }
}

File: src/Baekjoon/p10845.cpp
#include <iostream>
#include <string>

using namespace std;


// 전역변수 선언
int N, X;

// 큐 클래스 구현
class Queue {
    private:
        int container[10000];
        int frontIdx = 0; 
        int backIdx = 0;
    
    public:
        void push(int X) {
            container[backIdx++] = X;
        }

        int pop() {
            if (frontIdx == backIdx) return -1;
            else return container[frontIdx++];
        }

        int size() {
            return backIdx - frontIdx;
        }

        int empty() {
            if (frontIdx == backIdx) return 1;
            else return 0;
        }

        int front() {
            if (frontIdx == backIdx) return -1;
            else return container[frontIdx];
        }

        int back() {
            if (frontIdx == backIdx) return -1;
            else return container[backIdx-1];
        }
};

int main(void) {
    // 0. 입출력 시간 줄이기
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    // 1. 변수 선언
    string cmd = "";
    Queue queue = Queue();

    // 2. 입력받아서 명령 처리
    cin >> N;

    for (int i = 0; i < N; i++) {
        cin >> cmd;

        if (cmd == "push") {
            cin >> X;
            queue.push(X);
        }
        else if (cmd == "pop") {
            cout << queue.pop() << '\n';
        }
        else if (cmd == "size") {
            cout << queue.size() << '\n';
        }
        else if (cmd == "empty") {
            cout << queue.empty() << '\n';
        }
        else if (cmd == "front") {
            cout << queue.front() << '\n';
        }
        else if (cmd == "back") {
            cout << queue.back() << '\n';
        }
    }
}